### PR TITLE
Fix optionsWithParameter list

### PR DESCRIPTION
**Pull request by webmaster128**
This is a small set of patches to make #144 smaller. It fixes issue #143.

This is ready to merge as it is.

### CODE REVIEW

File: clcache.py [hunk 1]
```diff
@@ -792,9 +792,9 @@ def _parseOptionsAndFiles(cmdline):
             'Ob', 'Gs', 'Fa', 'Fd', 'Fm',
             'Fp', 'FR', 'doc', 'FA', 'Fe',
             'Fo', 'Fr', 'AI', 'FI', 'FU',
-            'D', 'U', 'I', 'Zp', 'vm',
+            'D', 'U', 'I', 'Zp',
             'MP', 'Tc', 'V', 'wd', 'wo',
-            'W', 'Yc', 'Yl', 'Tp', 'we',
+            'Yc', 'Yl', 'Tp', 'we',
```
Comment (frerich) [inline clcache.py]: I don't think this is correct for Visual Studio 2013. The [documentation](https://msdn.microsoft.com/en-us/library/thxezb7y(v=vs.120).aspx) explains that `/W` takes a numeric argument, e.g. both `/W4` as well as `/W 4` are valid.

File: clcache.py [hunk 2]
```diff
@@ -793,9 +793,9 @@ def _parseOptionsAndFiles(cmdline):
             'Fp', 'FR', 'doc', 'FA', 'Fe',
             'Fo', 'Fr', 'AI', 'FI', 'FU',
             'D', 'U', 'I', 'Zp',
-            'MP', 'Tc', 'V', 'wd', 'wo',
-            'Yc', 'Yl', 'Tp', 'we',
-            'Yu', 'Zm', 'F', 'Fi',
+            'F', 'Fi', 'MP', 'Tc', 'Tp', 'V',
+            'w1', 'w2', 'w3', 'w4', 'wd', 'we', 'wo', 'Wv',
```
Comment (frerich) [inline clcache.py]: A test for this would be nice which verifies that e.g. `['/w3', '1234']` is parsed correctly.

**Comment by frerich** [comment]
Looks promising - I think the fix for `/wn` which is apparently new in Visual Studio 2015 deserves a change log entry.

File: clcache.py (hunk 1 above)
Comment (webmaster128) [inline clcache.py]: Is `/W 4` really valid? I don't see any source for that. All documentation is talking about `/W4`. Is the space version used in the wild?

**Comment by webmaster128** [comment]
I push the test into a separate issue because I would like to do an interface change in CommandLineAnalyzer, that allows to do those tests. Right now, there is no getter to access those intermediate data and it is thrown away because unused in analyze().

But the implementation will handle it correctly, since when `/w3` prefix is found it looks for more parameter data in the argument (`/w3xxxxxxx`) and when there is none, it takes the next argument. So `['/w3', '1234']` as well as `['/w31234']` leads to `{'/w3': ['1234']}`.

`/wd`, `/we` and `/wo` are unchanged and just sorted alphabetically. So the commit log includes all added elements.

Since the parsed options are thrown away in analyze(), I don't think there is a real point for the changelog. The only thing that happens is that `cl /c /w1 123 main.cpp` will not read 123 and main.cpp as two different source files anymore. But I think `/w1 123` is wrong anyway and must be `/w1123` in both [MSVS 2008-2013](https://msdn.microsoft.com/en-us/library/thxezb7y(v=vs.120).aspx) (see /wln) and [MSVS 2015](https://msdn.microsoft.com/en-us/library/thxezb7y(v=vs.140).aspx).

**Comment by webmaster128** [comment]
I updated the PR. /W stays in and the `/w1 /w2 /w3 /w4 /Wv` commit is now smaller.

**Pull request merged**